File: client/src/app/model/free-entity-dto.ts
import {AddressDTO} from './address-dto.model';

export class FreeEntityDTO {
  public id!: number;
  public type!: string;
  public startDate!: Date;
  public endDate!: Date;
  public numberOfGuests!: number;
  public grade!: number;
  public country!: string;
  public city!: string;
  //public address: AddressDTO = new AddressDTO('', '', '', '');
}

File: isa2/demo/src/main/java/isa2/demo/DTO/FreeEntityDTO.java
package isa2.demo.DTO;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

@Getter
@Setter
public class FreeEntityDTO {
    private Integer numberOfGuests;
    private String type;
    @DateTimeFormat(iso = DateTimeFormat.ISO.TIME)
    private LocalDateTime startDate;
    @DateTimeFormat(iso = DateTimeFormat.ISO.TIME)
    private LocalDateTime endDate;
    //private AddressDTO address;
    private String country;
    private String city;
    private Double grade;
}
